CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLWorker.cc

```diff
@@ -0,0 +1,127 @@
+/****************************************************************************
+ *
+ * This is a part of PPS offline software.
+ * Authors:
+ *   Edoardo Bossini
+ *   Piotr Maciej Cwiklicki
+ *   Laurent Forthomme
+ *
+ ****************************************************************************/
+
+#include "DQMServices/Core/interface/DQMGlobalEDAnalyzer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "Geometry/VeryForwardGeometryBuilder/interface/CTPPSGeometry.h"
+#include "Geometry/Records/interface/VeryForwardRealGeometryRecord.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSDiamondDetId.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSDiamondRecHit.h"
+
+#include "CalibPPS/TimingCalibration/interface/TimingCalibrationStruct.h"
+
+//------------------------------------------------------------------------------
+
+class PPSTimingCalibrationPCLWorker : public DQMGlobalEDAnalyzer<TimingCalibrationHistograms> {
+public:
+  explicit PPSTimingCalibrationPCLWorker(const edm::ParameterSet&);
+
+  void dqmAnalyze(const edm::Event&, const edm::EventSetup&, const TimingCalibrationHistograms&) const override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  void bookHistograms(DQMStore::IBooker&,
+                      const edm::Run&,
+                      const edm::EventSetup&,
+                      TimingCalibrationHistograms&) const override;
+
+  edm::EDGetTokenT<edm::DetSetVector<CTPPSDiamondRecHit>> diamondRecHitToken_;
+  const std::string dqmDir_;
+};
+
+//------------------------------------------------------------------------------
+
+PPSTimingCalibrationPCLWorker::PPSTimingCalibrationPCLWorker(const edm::ParameterSet& iConfig)
+    : diamondRecHitToken_(
+          consumes<edm::DetSetVector<CTPPSDiamondRecHit>>(iConfig.getParameter<edm::InputTag>("diamondRecHitTag"))),
+      dqmDir_(iConfig.getParameter<std::string>("dqmDir")) {}
+
+//------------------------------------------------------------------------------
+
+void PPSTimingCalibrationPCLWorker::bookHistograms(DQMStore::IBooker& iBooker,
+                                                   const edm::Run& iRun,
+                                                   const edm::EventSetup& iSetup,
+                                                   TimingCalibrationHistograms& iHists) const {
+  iBooker.cd();
+  iBooker.setCurrentFolder(dqmDir_);
+  std::string ch_name;
+
+  edm::ESHandle<CTPPSGeometry> hGeom;
+  iSetup.get<VeryForwardRealGeometryRecord>().get(hGeom);
```

Comment:
@makortel I believe https://github.com/cms-sw/cmssw/pull/33263#issuecomment-806245684 was intended for this usage? Should I use a safer version of this record retrieval?

Reply:
I blatantly used this PR as a test bed for infrastructural PRs that should report more loudly in PR tests this way of getting EventSetup products that is deprecated. The registration of the consumed EventSetup products and getting data with the `ESGetToken` are documented in
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#In_ED_module
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_from_EventSetup_wit

Reply:
Thanks, @makortel ! Might be a sign of how long this PCL development has been ongoing on our side :)
Now fixed.